code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1172,7 +1170,7 @@
 	var/obscured = check_obscured_slots()
 
 	// If the eyes are covered by anything but glasses, that thing will be covering any potential glasses as well.
-	if(glasses && is_eyes_covered(ITEM_SLOT_MASK|ITEM_SLOT_HEAD) && glasses.wash(clean_types))
+	if(glasses && !(obscured & ITEM_SLOT_EYES) && glasses.wash(clean_types))
```

Comment:
why remove the verbose proc?

Reply:
this uses flags_cover instead of flags_inv, felt weird to use it there, i can revert that though

Reply:
also i think that check should be flipped either way? i think currently it cleans glasses if they are covered by something